com.ibm.wala.core/src/com/ibm/wala/classLoader/BytecodeClass.java

```diff
@@ -296,7 +296,8 @@ public IClass getSuperclass() {
       computeSuperclass();
     }
     if (superClass == null && !getReference().equals(TypeReference.JavaLangObject)) {
-      throw new NoSuperclassFoundException("No superclass found for " + this + " Superclass name " + superName);
+      //throw new NoSuperclassFoundException("No superclass found for " + this + " Superclass name " + superName);
```

Comment:
Why do we need this change?  What fails if we continue to throw the exception?  I'm hesitant about his change as this code has been throwing an exception for quite a while.

Reply:
It throws `NoSuperclassFoundException` while making CFG when the callee is in a `BytecodeClass` but has a `PhantomClass` superclass, as follows:

`com.ibm.wala.classLoader.NoSuperclassFoundException: No superclass found for <Application,Landroid/support/v4/view/AsyncLayoutInflater$BasicInflater> Superclass name Landroid/view/LayoutInflater
 at com.ibm.wala.classLoader.BytecodeClass.getSuperclass(BytecodeClass.java:299)
 at com.ibm.wala.classLoader.BytecodeClass.getMethod(BytecodeClass.java:464)
 at com.ibm.wala.classLoader.JavaLanguage.inferInvokeExceptions(JavaLanguage.java:585)
 at com.ibm.wala.cfg.ShrikeCFG$BasicBlock.addExceptionalEdges(ShrikeCFG.java:287)
 at com.ibm.wala.cfg.ShrikeCFG$BasicBlock.computeOutgoingEdges(ShrikeCFG.java:243)
 at com.ibm.wala.cfg.ShrikeCFG$BasicBlock.access$0(ShrikeCFG.java:232)
 at com.ibm.wala.cfg.ShrikeCFG.computeEdges(ShrikeCFG.java:133)
 at com.ibm.wala.cfg.ShrikeCFG.<init>(ShrikeCFG.java:76)
 at com.ibm.wala.cfg.ShrikeCFG.make(ShrikeCFG.java:63)
 at com.ibm.wala.classLoader.ShrikeIRFactory.makeCFG(ShrikeIRFactory.java:43)
 at com.ibm.wala.classLoader.ShrikeIRFactory.makeIR(ShrikeIRFactory.java:58)
 at com.ibm.wala.ssa.DefaultIRFactory.makeIR(DefaultIRFactory.java:69)`

Reply:
Ah ok; ugh.  Thanks for pointing this out.  I'd like to think a bit more about the right fix here.  It's possible I should just tweak the phantom class hacky implementation to work properly.